MapboxMobileEvents/MMEMetricsManager.m

```diff
@@ -0,0 +1,143 @@
+#import "MMEMetricsManager.h"
+#import "MMEEvent.h"
+#import "MMEReachability.h"
+#import "MMEConstants.h"
+
+@interface MMEMetricsManager ()
+
+@property (nonatomic) MMEMetrics *metrics;
+
+@end
+
+@implementation MMEMetricsManager
+
++ (instancetype)sharedManager {
+    static MMEMetricsManager *_sharedManager;
+    static dispatch_once_t onceToken;
+    
+    dispatch_once(&onceToken, ^{
+        _sharedManager = [[MMEMetricsManager alloc] init];
+        _sharedManager.metrics = [[MMEMetrics alloc] init];
+    });
+    
+    return _sharedManager;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        _metrics = [[MMEMetrics alloc] init];
+    }
+    return self;
+}
+
+- (void)metricsFromEventQueue:(NSArray *)eventQueue {
+    if (self.metrics.dateUTC == nil) {
```

Comment:
this block line 35-37 conflicts with the name of this method...

Reply:
Good point, moved this to a more appropriate place.